The project has been transferred to asynchronous methods.

WarFire5 opened the pull request.

WarFire5 committed:
The project has been transferred to asynchronous methods.

Negodyaj commented on TransactionStore.Business/Services/TransactionsService.cs:
```diff
@@ -107,23 +107,23 @@ public void AddTransferTransaction(TransferRequest request)
     }
 
-    public TransactionDto CreateWithdrawTransaction(TransferRequest request)
+    public async Task<TransactionDto> CreateWithdrawTransactionAsync(TransferRequest request)
```
этот метод не надо делать асинхронным
тут нет никаких долгих операций

Negodyaj commented on TransactionStore.Business/Services/TransactionsService.cs:
```diff
@@ -107,23 +107,23 @@ public void AddTransferTransaction(TransferRequest request)
     }
 
-    public TransactionDto CreateDepositTransaction(TransferRequest request)
+    public async Task<TransactionDto> CreateDepositTransactionAsync(TransferRequest request)
```
этот метод не надо делать асинхронным
тут нет никаких долгих операций

Negodyaj commented on TransactionStore.Core/Data/CurrencyRatesProvider.cs:
```diff
@@ -1,47 +1,46 @@
+    }
 
-        public decimal ConvertFirstCurrencyToUsd(Enum currencyEnum)
+    public Task<decimal> ConvertFirstCurrencyToUsdAsync(Enum currencyEnum)
```
этот метод не надо делать асинхронным
тут нет никаких долгих операций

Negodyaj commented on TransactionStore.Core/Data/CurrencyRatesProvider.cs:
```diff
@@ -1,47 +1,46 @@
+    }
 
-        public decimal ConvertUsdToSecondCurrency(Enum currencyEnum)
+    public Task<decimal> ConvertUsdToSecondCurrencyAsync(Enum currencyEnum)
```
этот метод не надо делать асинхронным
тут нет никаких долгих операций

WarFire5 committed:
Some asynchronous methods have been converted to regular methods.

WarFire5 committed:
Updated API tests. Simpled ExceptionMiddleware file.

Removed mapper and validator mocks from API tests. Simpled ExceptionMiddleware file.

WarFire5 committed:
Updated TransactionsRepositoryTests

WarFire5 committed:
Updated order of methods and tests. Clarified name of GetTransactionById

WarFire5 committed:
The status of post-methods has been changed to Created. Added new tests.
